### ISSUE TITLE
📎 Implement `noStaticElementId`

### ISSUE THREAD

**Issue by dyc3** [biomejs/biome#6036]
### Description

In HTML, you can assign elements a **globally unique** id via the `id` attribute. React has a utility hook, [`useId`](https://react.dev/reference/react/useId) to help generate ids that won't conflict. The primary purpose of this hook to to give elements IDs for a11y reasons.

This task is to implement a new javascript rule in the react domain to ban setting `id` on elements to a string literal.

Invalid:
```jsx
<div id="foo"></div>
```

Valid:
```jsx
const id = useId();
<div id={id} ></div>
```

This rule would be unique to Biome (as far as we know). The initial implementation can be very simple.

Discussed in #6025. We can bikeshed the name.

**Want to contribute?** Lets you know you are interested! We will assign you to the issue to prevent several people to work on the same issue. Don't worry, we can unassign you later if you are no longer interested in the issue! Read our [contributing guide](https://github.com/biomejs/biome/blob/main/CONTRIBUTING.md) and [analyzer contributing guide](https://github.com/biomejs/biome/blob/main/crates/biome_analyze/CONTRIBUTING.md).

**Comment by mehm8128** [comment]
Hi, it looks good rule. I'd like to try to implement it.

**Comment by dyc3** [comment]
All yours :)

**Comment by mehm8128** [comment]
FYI
I found a similar rule in Markuplint.
https://github.com/markuplint/markuplint/tree/dev/packages/%40markuplint/rules/src/no-hard-code-id
